tests/unit/components/enhanced-table/enhanced-table-head/EnhancedTableHead.spec.jsx

```diff
@@ -0,0 +1,54 @@
+import { vi } from 'vitest'
+import { render, screen, fireEvent } from '@testing-library/react'
+import { TableProvider } from '~/context/table-context'
+import EnhancedTableHead from '~/components/enhanced-table/enhanced-table-head/EnhancedTableHead'
+
+const columnsMock = [
+  {
+    label: 'Name',
+    field: 'name',
+    dataType: 'string',
+  },
+  {
+    label: 'Email',
+    field: 'email',
+    dataType: 'string'
+  },
+]
+const initialSortMock = { order: 'asc', orderBy: 'email' }
+const itemsCountMock = 10
+const onSelectAllClickMock = vi.fn()
+
+describe('EnhancedTableHead', () => {
+  beforeEach(()=>{
+    render(
+      <TableProvider columns={ columnsMock } initialSort={ initialSortMock } isSelection>
+        <table>
+          <EnhancedTableHead
+            itemsCount={ itemsCountMock }
+            onSelectAllClick={ onSelectAllClickMock }
+          />
+        </table>
+      </TableProvider>
+    )
+  })
+
+  it('renders all columns in the table', () => {
+    const nameColumn = screen.getByText(columnsMock[0].label)
+    const emailColumn = screen.getByText(columnsMock[1].label)
```

Comment:
Is it necessary to check name and emai? maybe one of them would be enough

Reply:
no, it is not necessary, but I have two columns in mock. decided to check both